agreements/notePurchaseAgreementsGateway.go

```diff
@@ -65,22 +68,26 @@ func (g *NotePurchaseAgreementGateway) SaveNotePurchaseAgreement(ctx context.Con
 	return agreement, nil
 }
 
-func (g *NotePurchaseAgreementGateway) Getall(ctx context.Context) ([]interfaces.IAgreementModel, error) {
-	return nil, nil
-	// results := g.repository.Query(nil)
-	// models := []interfaces.IAgreementModel{}
+func (g *NotePurchaseAgreementGateway) Getall(ctx context.Context) ([]*interop.NotePurchaseAgreement, error) {
+	agreements := make([]*interop.NotePurchaseAgreement, 0)
 
-	// for _, result := range results {
-	// 	model, err := g.factory.NewPurchaseAgreementModel(ctx, result)
+	usrID := fmt.Sprintf("%v", ctx.Value(constants.UserIdKey))
```

Comment:
I see no problem with what you did in the interceptor.  It makes sense to make the user info available to the controller, as well as the client, but the gateway layer is the wrong layer to retrieve the info.  This userid should be accessed in the controller layer, and passed as an argument (or in a DTO) to the service/gateway layer.

Reply:
That makes sense. I will update the userId to be passed from the controller layer. I think the userId can still remain in the context, in order to enrich to logs and metrics information in the future.